### ISSUE TITLE
Subclassing WebComponents and calling super().__init__()?

### ISSUE THREAD

**Issue by AgarwalPragy** [brython-dev/brython#1163]
```html
    <child-component></child-component>
    <script type="text/python">
        from browser.webcomponent import WebComponent
        from browser.html import B

        class Parent(metaclass=WebComponent):
            tag_name = 'parent-component'

            def __init__(self):
                shadow = self.attachShadow({'mode': 'open'})
                shadow <= B('parent component')

        class Child(Parent):
            tag_name = 'child-component'

            def __init__(self):
                super().__init__()
                shadow = self.attachShadow({'mode': 'open'})
                shadow <= B('child component')
    </script>
```

Calling `super().__init__()` in the Child's `__init__()` method results in the following traceback

```
TypeError: Cannot read property '__mro__' of undefined
    at $$super.__getattribute__ (brython.js:7030)
    at Object.$B.$getattr (brython.js:6587)
    at __init__1 (eval at $B.loop (brython.js:5108), <anonymous>:136:29)
    at brython.js:8546
    at HTMLElement.__init__ (brython.js:13088)
    at new WebComp (brython.js:13087)
    at Object.$B.createWebComponent (brython.js:13089)
    at f (brython.js:6516)
    at __new__0 (eval at exec_module (brython.js:8861), <anonymous>:51:72)
    at Object.$B.$class_constructor (brython.js:5388)
brython.js:6520 Uncaught Error
    at Object._b_.AttributeError.$factory (eval at $make_exc (brython.js:7501), <anonymous>:41:354)
    at Object.$B.$getattr (brython.js:6520)
    at Object.getattr (brython.js:6484)
    at brython.js:8547
    at HTMLElement.__init__ (brython.js:13088)
    at new WebComp (brython.js:13087)
    at Object.$B.createWebComponent (brython.js:13089)
    at f (brython.js:6516)
    at __new__0 (eval at exec_module (brython.js:8861), <anonymous>:51:72)
    at Object.$B.$class_constructor (brython.js:5388)
```

**Comment by PierreQuentel** [comment]
Thanks for reporting this issue. I tried to fix it without changing the API, but realized that there was something wrong with the current implementation : the binding between tag name and Python class was made _before_ the class was actually created (in the metaclass method `__new__`), which would work in some cases, but not when a reference to the class is used in `__init__`, which is the case with `super()`.

In the new version introduced by the above commit, the module __`browser.webcomponent`__ only exposes one function, `define`, which is used to bind the tag name to a class ; this class no longer needs a metaclass, nor a `tag_name` attribute.

With this change, your example would be: 
```python
from browser import webcomponent
from browser.html import B, BR

class Parent:

    def __init__(self):
        self.shadow = self.attachShadow({'mode': 'open'})
        self.shadow <= B('parent component') + BR()

class Child(Parent):

    def __init__(self):
        super().__init__()
        self.shadow <= B('child component')

    def connectedCallback(self):
        print("connected")

webcomponent.define("parent-component", Parent)
webcomponent.define("child-component", Child)
```

Note that the shadow element defined in the `Parent` class is stored as an attribute of `self` and reused in `Child.__init__()`, you cannot call `attachShadow()` a second time on the element, I think this is because of constraints on the `ShadowRoot` object.

**Comment by AgarwalPragy** [comment]
Can you please check this again?

I'm getting the following traceback when trying to create a child component

```js
TypeError: Cannot read property 'shadow' of undefined
    at JSObject.__getattribute__ (brython.js:8635)
    at Object.$B.$getattr (brython.js:6638)
    at __init__1 (eval at $B.loop (brython.js:5146), <anonymous>:136:42)
    at new WebComponent (eval at run_js (brython.js:8806), <anonymous>:42:39)
    at define (eval at run_js (brython.js:8806), <anonymous>:61:20)
    at eval (eval at $B.loop (brython.js:5146), <anonymous>:240:69)
    at $B.loop (brython.js:5146)
    at $B.inImported (brython.js:5136)
    at $B.loop (brython.js:5152)
    at idb_load (brython.js:5079)
```

**Comment by PierreQuentel** [comment]
Sorry about that :-( The regression happened when I fixed issue #1203
